unittests/test_policies.py

```diff
@@ -262,32 +262,35 @@ def test_dependencies(self):
 
     def test_sigterm(self):
         # Wrapper of self.runall which is used from a child process and
-        # passes any exception to the parent process
-        def _runall(checks, conn):
+        # passes any exception, number of cases and failures to the parent
+        # process
+        def _runall(checks, ns):
             exc = None
             try:
                 self.runall(checks)
             except BaseException as e:
                 exc = e
-
-            stats = self.runner.stats
-            conn.send((exc, stats.num_cases(), len(stats.failures())))
-            conn.close()
-
-        rd_endpoint, wr_endpoint = multiprocessing.Pipe(duplex=False)
-        p = multiprocessing.Process(target=_runall,
-                                    args=([SleepCheck(3)], wr_endpoint))
-        p.start()
-
-        # The unused write endpoint has to be closed from the parent to
-        # ensure that the `recv()` method of `rd_endpoint` returns
-        wr_endpoint.close()
-
-        # Allow some time so that the SleepCheck is submitted
-        time.sleep(1)
-        p.terminate()
-        p.join()
-        exc, num_cases, num_failures = rd_endpoint.recv()
+            finally:
+                ns.exc = exc
+                ns.num_cases = self.runner.stats.num_cases()
+                ns.num_failures = len(self.runner.stats.failures())
+
+        with multiprocessing.Manager() as manager:
+            ns = manager.Namespace()
+            p = multiprocessing.Process(target=_runall,
+                                        args=([SleepCheck(20)], ns))
+
+            p.start()
+
+            # Allow some time so that the SleepCheck is submitted.
+            # The sleep time of the submitted test is much larger to
+            # ensure that it does not finish before the termination signal
+            time.sleep(0.2)
+            p.terminate()
+            p.join()
+            exc = ns.exc
+            num_cases = ns.num_cases
+            num_failures = ns.num_failures
```

Comment:
Remove these three lines and work directly with the namespace object in the following.

Reply:
These fields are not accessible outside of the context of the manager that created the `Namespace` object. I will put the `assert` inside the `with` as discussed.